### PR TITLE
OpenGraph: changed the wpvideo link to use the mp4 video.

### LINKED ISSUE #2950: Open Graph: Stop using Flash for og:video
Right now, we have Flash hardcoded: https://github.com/Automattic/jetpack/blob/8a30daea4c9caea855c24b0ea7e9758d35e01e32/modules/publicize/enhanced-open-graph.php#L89

Per Otto, FB accepts mp4, which we should try to use, at least for VideoPress.

Would fix: https://meta.trac.wordpress.org/ticket/1375

### PR DESCRIPTION
**Pull request by zinigor**
This fixes #2950.

### CODE REVIEW

**Comment by jeherve** [comment]
That won't work I'm afraid. Facebook expects the `video` tag to include a direct link to an mp4 file, and not just a video permalink. If you only use the video permalink, you get the following markup when sharing the post on Facebook:

screen shot 2015-11-09 at 2 05 43 pm

And the video won't play:

screen shot 2015-11-09 at 2 07 18 pm

Although VideoPress includes a direct mp4 file link, I'm not how we can build it in Jetpack. Here is the usual link format:
`https://videos.files.wordpress.com/EI79343G/kauai-aurora-meetup_hd.mp4`

We know the VideoPress ID, but I'm not sure how we can get the file name.

cc @kraftbj who's looked into this as well.

**Comment by zinigor** [comment]
@jeherve you're right, thanks for pointing that out. I wanted to avoid using the `VideoPress_Video` class, but it looks like so far it's the only way. The tradeoff is that we have to make a synchronous API call to VideoPress, but the bonus is that now we are able to use the poster URL!

**Comment by kraftbj** [comment]
Haven't tested the branch. I notice the VP class `isset` checks before setting many/most of the variables. We probably should have a fallback to check is any particular one is set (e.g. `$wpvideo->videos->mp4->url` ) and output sane OG tags still in that case without throwing any notices.

File: class.media-summary.php [hunk 1]
```diff
@@ -58,10 +58,15 @@ static function get( $post_id, $blog_id = 0, $args = array() ) {
...
+								require_once( JETPACK__PLUGIN_DIR . '/modules/videopress/class.videopress-video.php' );
+							}
+							$wpvideo = new VideoPress_Video( $extract['shortcode']['wpvideo']['id'][0] );
+
 							$return['type'] = 'video';
-							$return['video'] = esc_url_raw( 'http://s0.videopress.com/player.swf?guid=' . $extract['shortcode']['wpvideo']['id'][0] . '&isDynamicSeeking=true' );
-							$return['image'] = self::get_video_poster( 'videopress', $extract['shortcode']['wpvideo']['id'][0] );
+							$return['video'] = esc_url_raw( $wpvideo->videos->mp4->url );
+							$return['image'] = $wpvideo->poster_frame_uri;
 							$return['secure']['video'] = preg_replace( '@http://[^\.]+.videopress.com/@', 'https://v0.wordpress.com/', $return['video'] );
```
Comment (kraftbj) [inline class.media-summary.php]: We need to change this to the mp4 as well, ensuring the scheme is set to https.

Comment (zinigor) [inline class.media-summary.php]: Not sure what you mean here. We are using the new URL and swapping out the `http` part with `https` one.

**Comment by zinigor** [comment]
@kraftbj thanks for pointing out the problem! I have dealt with it in the latest commit. One thing that came to my mind is that we can put the video caching code into the Video class too, but that's outside the scope of this issue.

**Comment by samhotchkiss** [comment]
This seems like a lot of moving pieces for a point release-- let's punt to 3.9

**Comment by kraftbj** [comment]
The og:video:type tag is still set to `application/x-shockwave-flash`, which needs to be set to `video/mp4`.

**Comment by zinigor** [comment]
@kraftbj can you please provide a reproduction scenario? I have handled the type here, very curious to see where it fails:
https://github.com/Automattic/jetpack/pull/2980/files#diff-65ba43837aeb17dc2f17fc7717497c6aR104

**Comment by kraftbj** [comment]
@zinigor Use a shortcode. `[wpvideo OcobLTqC]` still uses the flash in the og tags.

**Comment by zinigor** [comment]
@kraftbj just returned to this PR, sorry for the delay. When I'm using the shortcode that you have provided, I still get `mp4`:

```
<meta property="og:video:type" content="video/mp4" />
<meta property="og:video" content="http://videos.videopress.com/OcobLTqC/img_5786_dvd.mp4" />
<meta property="og:video:secure_url" content="https://v0.wordpress.com/OcobLTqC/img_5786_dvd.mp4" />
```

**Comment by jeherve** [comment]
Closing in favor of #3853

**Pull request closed without merging**